lib/ToolbarPanelHub.jsm

```diff
@@ -34,12 +34,25 @@ class _ToolbarPanelHub {
     this._getMessages = getMessages;
     if (this.whatsNewPanelEnabled) {
       this.enableAppmenuButton();
+      // Listen for pref changes that could turn off the feature
+      Services.prefs.addObserver(WHATSNEW_ENABLED_PREF, this);
```

Comment:
This should happen outside of the `if` statement as the pref will be `false` by default.